src/components/data_submission/DataAccessGovernance.js

```diff
@@ -40,22 +94,92 @@ export const DataAccessGovernance = (props) => {
   const addNewConsentGroup = useCallback(() => {
     setConsentGroupsState((consentGroupsState) => {
       const newConsentGroupsState = cloneDeep(consentGroupsState);
-      newConsentGroupsState.push({
+      newConsentGroupsState?.push({
         consentGroup: {},
         key: Math.max(0, ...consentGroupsState.map((state) => state.key)) + 1,
         nihInstitutionalCertificationFile: null,
         editMode: true,
         valid: false,
+        disableFields: false,
       });
 
       return newConsentGroupsState;
     });
   }, []);
 
+  // extract consent groups from datasets
+  const prefillConsentGroups = useCallback(async () => {
+    // from datasets array to consent group array (needs to match the existing consent group structure)
+    var consentGroups = await Promise.all(datasets?.map(async (dataset, idx) => {
+      const dataUse = await normalizeDataUse(dataset?.dataUse);
+      // DAC is required if openAcess is false
+      const openAccess = extract('Open Access', dataset);
+      const dac = openAccess ? undefined : await DAC.get(dataset?.dacId);
+      const dacs = openAccess ? undefined : await DAC.list();
+
+      // extract the consent group data from the dataset
+      return {
+        consentGroup: {
+          consentGroupName: dataset.name,
+
+          // primary
+          generalResearchUse: dataUse.generalUse,
+          hmb: dataUse.hmbResearch,
+          diseaseSpecificUse: dataUse.diseaseRestrictions,
+          poa: dataUse.populationOriginsAncestry,
+          openAccess: openAccess,
+          otherPrimary: dataUse.other,
+
+          // secondary
+          nmds: dataUse.methodsResearch,
+          gso: dataUse.geneticStudiesOnly,
+          pub: dataUse.publicationResults,
+          col: dataUse.collaboratorRequired,
+          irb: dataUse.ethicsApprovalRequired,
+          gs: dataUse.geographicalRestrictions,
+          mor: dataUse.publicationMoratorium,
+          npu: dataUse.commercialUse,
+          otherSecondary: dataUse.secondaryOther,
+          url: extract('URL', dataset),
+          dataLocation: extract('Data Location', dataset),
+          numberOfParticipants: extract('Number Of Participants', dataset),
+          fileTypes: [
+            {
+              fileType: extractFileTypes('File Types', 'fileType', dataset),
+              functionalEquivalence: extractFileTypes('File Types', 'functionalEquivalence', dataset)
+            }
+          ],
+          dataAccessCommitteeId: { ...dac, dacs },
+        },
+        key: `prefilledConsentGroups[${idx}]`,
+        nihInstitutionalCertificationFile: null,
+        editMode: false,
```

Comment:
Same question with `editMode`, except the situation would be `const [editMode, setEditMode] = useState(!studyEditMode);`